optimum/neuron/modeling_base.py

```diff
@@ -154,6 +156,16 @@ def _from_pretrained(
             else:
                 file_name = neuron_files[0].name
 
+        # check compiler compatability of saved model vs system installation
+        # Vagrantly assumes that the user won't try to load an xcc on a cc machine (and vise/versa)
+        if hasattr(config, "neuron_compiler") and hasattr(config, "neuron_compiler_version"):
+            min_version = get_neuronxcc_version() if config.neuron_compiler == "neuronx-cc" else get_neuroncc_version()
+            if version.parse(config.neuron_compiler_version) > version.parse(min_version):
```

Comment:
Here I am curious about the backward compatibility of the neuron compiler. Would a model compiled with an older version of neuron(x)-cc compiler be compatible to run under a newer environment?  @mmcclean-aws

Reply:
NEFFs are backwards compatible, but official confirmation would be good from @mcclean-aws .

Reply:
Yes, the Neuron runtime ensures backwards compatibility of NEFFs